crates/tests/src/e2e/ibc_tests.rs

```diff
@@ -384,6 +387,53 @@ fn ibc_transfers() -> Result<()> {
     check_balance(&test, AA_VIEWING_KEY, APFEL, 0)?;
     check_balance(&test, IBC_REFUND_TARGET_ALIAS, APFEL, 1)?;
 
+    // 6. Malformed shielded actions
+
+    // Check initial balance
+    check_balance(&test, AA_VIEWING_KEY, &ibc_denom_on_namada, 40)?;
+
+    // Missing memo
+    transfer_from_gaia(
+        &test_gaia,
+        GAIA_USER,
+        AA_PAYMENT_ADDRESS,
+        GAIA_COIN,
+        100,
+        &port_id_gaia,
+        &channel_id_gaia,
+        None,
+        // MASP VP shall reject it, make it timeout
+        Some(Duration::new(10, 0)),
+    )?;
+    wait_for_packet_relay(&port_id_namada, &channel_id_namada, &test)?;
+    // Check the balance didn't change
+    check_balance(&test, AA_VIEWING_KEY, &ibc_denom_on_namada, 40)?;
```

Comment:
Should we also check the gaia source balance to be refunded? In the actual issue case, the receiver balance wasn't changed even though the source balance was decreased.
```
check_gaia_balance(&test_gaia, GAIA_USER, GAIA_COIN, 810)?;
```

Reply:
That would be great yes. I've tried myself but I couldn't make the test pass (I kept seeing a balance of 710): would you have some time to try looking into it? I'm not very familiar with the relayer mechanism and our tests

Reply:
Thanks! 🙏🏻

Reply:
In my local, the checking the Gaia balance passed. The CI also passed without errors.